environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -662,6 +664,17 @@ public Weld addServices(Service... services) {
     }
 
     /**
+     * Sets the bean discovery mode for synthetic bean archive. Default mode is ALL.
+     * @param mode Bean discovery mode as string, possible values are ALL, ANNOTATED, NONE
+     *
+     * @return  self
+     */
+    public Weld setSyntheticArchiveBeanDiscoveryMode(String mode) {
```

Comment:
I think `setBeanDiscoveryMode()` should be ok. We don't use `syntheticArchive` in other methods neither.